air/tests/test_module/features/streams/recursive_streams.rs

```diff
@@ -470,3 +472,29 @@ fn recursive_stream_fold_with_n_service_call() {
 
     assert_eq!(actual_fold_state.lore.len(), expected_fold_lores);
 }
+
+#[test]
```

Comment:
BTW, it worth having a test where two datas that have, say, 1000 common stream elements and 1-2 additional new elements in current data.

Reply:
Resolved after a side discussion.